src/libs/TransactionUtils.ts

```diff
@@ -520,6 +521,13 @@ function getRecentTransactions(transactions: Record<string, string>, size = 2):
         .slice(0, size);
 }
 
+/**
+ * Checks if any violations for the provided transaction are of type 'violation'
+ */
+function hasViolation(transactionID: string, transactionViolations: OnyxCollection<TransactionViolation[]>): boolean {
```

Comment:
Q: why does this need to accept the violations as a 2nd parameter? Can't we subscribe to the key `ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS` similar to how we're subscribing to all transactions?

Reply:
> similar to how we're subscribing to all transactions

We don't recommend this because of [Problem](https://github.com/Expensify/App/issues/27262), though there're many places throughout the app.

Reply:
I don't see how that's related, the solution recommends using `Onyx.connect`?

>Switch all references to properly use withOnyx() for components and connect() for libs.

Reply:
bump @situchan

Reply:
The solution recommends withOnyx in view component, instead of Onyx.connect in static file

Reply:
Double check please. It's recommending `Onyx.connect` for libs, and `withOnyx` for views. `TransactionUtils` is a lib and it should be okay for it to subscribe to the violations key.

Reply:
`data is coming from somewhere that is not props or state and cannot be debugged in react dev tools` - this is the problem
Also, please check [this discussion](https://expensify.slack.com/archives/C01GTK53T8Q/p1688630003862989)

Reply:
Ah I see, I think it's an issue specifically with `useMemo`...I agree this can cause reactivity issues. Thanks for linking that.